sector-base/src/api/porep_config.rs

```diff
@@ -9,11 +9,14 @@ pub enum PoRepConfig {
     Test,
 }
 
+// When modifying, update internal::tests::partition_layer_challenges_test to reflect supported PoRepProofPartitions.
 #[derive(Clone, Copy, Debug)]
 pub enum PoRepProofPartitions {
     Two,
 }
 
+pub const POREP_PROOF_PARTITION_CHOICES: [PoRepProofPartitions; 1] = [PoRepProofPartitions::Two];
```

Comment:
SUGGESTION: I don't think that you should do this. I created an enum so that the compiler could enforce exhaustiveness of pattern matches - and this array defeats that. I think that instead of putting each enum member into a fixed length array and iterating over that array's elements that you should instead pattern match. This will prevent you/me from needing to maintain the `POREP_PROOF_PARTITION_CHOICES` constant.

Reply:
Yes, I considered that. It's not clear to me that you can use wildcard pattern matching to enumerate the choices of an enum — which is what we'd like to be able to do. Especially when enumerating combinations, a nested loop will work, but I think we'd need to duplicate whatever manual/matching enumeration strategy was chosen — in each branch. I could be missing something, though.

Reply:
After thinking about this again, I retract my original objection. If your intent is to generate parameters for each combination of sector size and PoRep proof partitions, then this is a fine approach. Alternatively, you could try to generate an `Iter` impl (see [here](https://cbreeden.github.io/Macros11/)).